### PR TITLE
Add purchase product barcode

### PR DESCRIPTION
**Pull request by TimLai125**

### CODE REVIEW

File: purchase_product_barcode/models/purchase_order_line.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+
+class PurchaseOrderLine(models.Model):
+    _inherit = 'purchase.order.line'
+
+    barcode = fields.Char(
+        related='product_id.barcode',
+        string='Barcode',
+        copy=False,
+    )
```
Comment (TimLai125) [inline purchase_product_barcode/models/purchase_order_line.py]: Considering about readonly attribute to this field.

File: purchase_product_barcode/reports/purchase_order_templates.xml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+        <xpath expr="//table[@class='table table-condensed']/thead/tr/th[1]"
+                   position="after">
+            <th><strong>Barcode</strong></th>
+        </xpath>
+        <xpath expr="//table[@class='table table-condensed']/tbody/tr/td[1]"
+                   position="after">
+            <td>
+                <span t-if="line.barcode">
+                    <img t-att-src="'/report/barcode/?type=%s&amp;value=%s&amp;width=%s&amp;height=%s'
+                                    % ('EAN13', line.barcode, 600, 100)"
```
Comment (TimLai125) [inline purchase_product_barcode/reports/purchase_order_templates.xml:14]: Odoo does not support 'JAN' barcode directly but 'JAN' is compatible with 'EAN13'.

**Comment by TimLai125** [comment]
The PR is good to review.

File: purchase_product_barcode/models/purchase_order_line.py (hunk 1 above)
Comment (yostashiro) [reply purchase_product_barcode/models/purchase_order_line.py]: Yes, please make the field readonly.

**Review by yostashiro — CHANGES_REQUESTED** [review]
@TimLai125 please merge this PR once you have made an update on barcode field.  Thanks!

**Comment by TimLai125** [comment]
@yostashiro I do not have the write access to this repository to merge the PR

**Comment by yostashiro** [comment]
> I do not have the write access to this repository to merge the PR

@TimLai125 Try again.  I have updated the team settings.

**Pull request merged**